pkg/agent/multicast/mcast_controller_test.go

```diff
@@ -452,6 +671,36 @@ func TestProcessPacketIn(t *testing.T) {
 	}
 }
 
+func TestHandlePacketInWithErrorCases(t *testing.T) {
+	mockController := newMockMulticastController(t, false)
+	snooper := mockController.igmpSnooper
+
+	for _, tc := range []struct {
+		pktIn ofctrl.PacketIn
+		err   error
+	}{
+		{
+			pktIn: generatePacketWithMatches(protocol.NewIGMPv3Report(make([]protocol.IGMPv3GroupRecord, 0)), 3, nil, []openflow15.MatchField{*openflow15.NewInPortField(3), {
+				Class:   openflow15.OXM_CLASS_PACKET_REGS,
+				Field:   openflow15.OXM_FIELD_IN_PORT,
+				HasMask: true,
+				Mask:    &openflow15.ByteArrayField{Data: []byte{255, 255, 255, 255, 0, 0, 0, 0}, Length: 8},
+				Value:   &openflow15.ByteArrayField{Data: []byte{0, 0, 0, 0, 0, 0, 0, 0}, Length: 8},
+			}}),
+			err: nil,
+		},
+		{
+			pktIn: generatePacketWithMatches(protocol.NewIGMPv3Report(make([]protocol.IGMPv3GroupRecord, 0)), 3, nil, []openflow15.MatchField{*openflow15.NewInPortField(3)}),
+			err:   fmt.Errorf("error getting match from IGMP marks in CustomField"),
+		},
+	} {
+		err := snooper.HandlePacketIn(&tc.pktIn)
+		if err != nil {
+			assert.Equal(t, err.Error(), tc.err.Error())
```

Comment:
No need to check with `if err != nil`, just use assert.Equal(t, err, tc.err)?

Reply:
updated